intellij/src/saros/intellij/editor/EditorManager.java

```diff
@@ -1023,21 +1012,7 @@ private void executeInUIThreadSynchronous(Runnable runnable) {
 
   @Override
```

Comment:
What is the contract in the interface ?

Reply:
For [`EditorManager.saveEditors(IProject)`](https://github.com/saros-project/saros/blob/master/core/src/saros/editor/IEditorManager.java#L79):
Save open editors of given project or all editors if given project is `null`. But this is based on the notion that only open editors can have modified content, which is not the case for IntelliJ. As mentioned in the commit message, documents (i.e. the editor content) can remain unsaved (i.e. are not written to disk) even if the editor is closed. The cached state will again be used when a new editor is opened for the document.

For [`DocumentAPI.saveAllDocuments()`](https://github.com/saros-project/saros/blob/master/intellij/src/saros/intellij/editor/DocumentAPI.java#L89):
Calls [`FileDocumentManager.saveAllDocuments()`](https://github.com/JetBrains/intellij-community/blob/master/platform/core-api/src/com/intellij/openapi/fileEditor/FileDocumentManager.java#L70) on the EDT, de-normalizing and then flushing all editor/document content to disk.